provider/github/watcher_test.go

```diff
@@ -46,11 +47,16 @@ func (s *WatcherTestSuite) SetupTest() {
 }
 
 func (s *WatcherTestSuite) TestWatch() {
-	var callsA, callsB, events, prEvents, pushEvents int
+	var callsA, callsB, events, prEvents, pushEvents int32
 
-	pullsHandler := func(calls *int) func(w http.ResponseWriter, r *http.Request) {
+	var mutex = &sync.Mutex{}
+
+	pullsHandler := func(calls *int32) func(w http.ResponseWriter, r *http.Request) {
 		return func(w http.ResponseWriter, r *http.Request) {
+			mutex.Lock()
```

Comment:
Maybe it would be easier to use `atomic` package here: `atomic.AddUint32(&calls, 1)`
And later when we read it `atomic.LoadUint32(&callsA)`

Reply:
ok, pretty similar